docs/openapi.yaml

```
@@ -0,0 +1,1143 @@
+openapi: 3.0.1
+info:
+  title: "Flotilla API"
+  version: "1.0.0"
+  contact:
+    email: "fg_robot_dev@equinor.com"
+
+  license:
+    name: Eclipse 2.0
+    url: https://www.eclipse.org/org/documents/epl-2.0/EPL-2.0.txt
+
+  description: |
+    The Flotilla API is an open API covering the robots in use on an asset within Equinor.
+    This document describes release v1.0.0. We recommend using documentation published at [github.io](https://equinor.github.io/api-first-workbench) as it provides the best developer experience.
+    # Introduction
+    The API is designed and built according to [Equinor's API strategy](https://github.com/equinor/api-strategy).
+    The API is documented in OpenAPI 3.0 format and follows REST principles.
+
+    ## API versioning
+    The API follows the principles of [API evolution](https://apisyouwonthate.com/blog/api-evolution-for-rest-http-apis)
+
+    ## Authentication and authorization overview
+    The user executing the API requests will need to authenticate with OAuth 2.0 authorization code flow with Equinor Azure AD as the identity provider. 
+
+    ## Dates
+    All date and time information are based on UTC (Universal Time Coordinated) and formatted according to [RFC3339](https://www.ietf.org/rfc/rfc3339.txt)
+    This also applies to durations, such as 1 day or 2 hours, which are encoded as `P0Y0M1DT0H0M0S` and `P0Y0M0DT2H0M0S`.
+
+    ## Deprecation policy
+    The API will use the principles of [API evolution](https://apisyouwonthate.com/blog/api-evolution-for-rest-http-apis) instead of introducing a new api-version on every breaking change.
+
+    In order to achieve this, endpoints, query parameters and resource properties can become deprecated and will be marked so in the OpenAPI specification.
+
+    ## Endpoints with deprecation
+    None
+    # Feature requests
+    Feature requests are welcomed and can be submitted by creating issues on: [GitHub](https://github.com/equinor/flotilla)
+
+externalDocs:
+  description: API documentation
+  url: "https://equinor.github.io/api-first-workbench"
+
+security:
+  - OAuth2:
+      - read
+      - write
+
+tags:
+  - name: Robots
+    description: |
+      Robots represent all robots on asset
```

```suggestion
      Robots represent all robots on the asset
```

Does this mean that we will have one Flotilla instance running per assest, i.e. is one Flotilla instance limited to serve one asset? This might be a design choice we have made, but I am just wondering.

---

Is it then an idea to have an endpoint that returns which asset the API is serving?

---

Agreed, that could be relevant

---

I think we agreed to have one instance of Flotilla per asset and one Flotilla serves only one asset since Flotilla is going to live somewhere within the firewalls of the asset. Knowing which asset the instance is living on should be a part of the configuration.

---

Good, that probably is sufficient